msg.go

```diff
@@ -62,13 +65,24 @@ var (
 	ErrNoSig         error = &Error{err: "no signature found"}
 	ErrPrivKey       error = &Error{err: "bad private key"}
 	ErrRcode         error = &Error{err: "bad rcode"}
-	ErrRdata         error = &Error{err: "bad rdata"}
 	ErrRRset         error = &Error{err: "bad rrset"}
 	ErrSecret        error = &Error{err: "no secrets defined"}
 	ErrShortRead     error = &Error{err: "short read"}
 	ErrSig           error = &Error{err: "bad signature"} // ErrSig indicates that a signature can not be cryptographically validated.
 	ErrSoa           error = &Error{err: "no SOA"}        // ErrSOA indicates that no SOA RR was seen when doing zone transfers.
 	ErrTime          error = &Error{err: "bad time"}      // ErrTime indicates a timing error in TSIG authentication.
+
+	// TODO(tmthrgd): Stop returning and deprecate ErrRdata. It's not well
+	// defined.
+	ErrRdata error = &Error{err: "bad rdata"}
+
+	// For CoreDNS, we need to use an error that either matches
+	// errors.Is(_, ErrBuf) or contains the substring "overflow".
+	//
+	// TODO(tmthrgd): Update CoreDNS and use "truncated message".
+	errTruncatedMessage = &Error{err: "overflow unpacking truncated message"}
```

Comment:
a properly truncated message should unpack just fine.
unsure why TC bit messages deserve special treatment?

Reply:
This isn't for a properly framed messaged with or without the TC bit set, this is for a message that is say randomly cut in half. We return this if we're say unpacking a resource, but the message buffer stops half way through the RDATA. This already fails, but with a less obvious error message. I'm open to suggestions on a different message.